### PR TITLE
[WIP] MVP of Kad inspection tools

### PR DESCRIPTION
**Pull request by dylanlott**
This is a WIP PR to introduce some tooling for inspecting, debugging, and modifying kademlia from a CLI as a secondary tool to help with debugging the overlay and kademlia systems.

### CODE REVIEW

File: cmd/kad/main.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+)
+
+// Inspector gives access to kademlia and overlay cache
+type Inspector struct {
+	overlay  overlay.Client
+	identity *provider.FullIdentity
+}
+
+func identity() (*provider.FullIdentity, error) {
+	ca, err := provider.NewTestCA(context.Background())
```
Comment (jenlij) [inline cmd/kad/main.go]: why do we use a testing ca?

File: cmd/kad/main.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	identity, err := ca.NewIdentity()
+	if err != nil {
+		zap.S().Errorf("Failed to create full identity: ", zap.Error(err))
+		os.Exit(1)
+	}
+	return identity, nil
+}
+
+// overlay returns an overlay client
+func newOverlayClient(identity *provider.FullIdentity, address string) (overlay.Client, error) {
```
Comment (wthorp) [inline cmd/kad/main.go]: What the benefit of wrapping a single call here?

File: pkg/pb/kadcli.proto [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+// See LICENSE for copying information.
+
+syntax = "proto3";
+option go_package = "pb";
+
+package kadcli;
+
+service KadCli {
+  // CountNodes returns the number of nodes in the cache and in the routing table
+  rpc CountNodes(CountNodesRequest) returns (CountNodesResponse);
```
[the remark this answers is not in the document]
Comment (dylanlott) [reply pkg/pb/kadcli.proto]: Not yet, but it will be later on.

File: cmd/kad/main.go (hunk 2 above)
Comment (dylanlott) [reply cmd/kad/main.go]: Originally it did some more setup for the client, this was an artifact of some refactoring, so I'll pull it out.

File: cmd/kad/main.go (hunk 1 above)
Comment (dylanlott) [reply cmd/kad/main.go]: I swapped this for `node.NewFullIdentity`

**Pull request merged**